### PR TITLE
Package types, providers and tests

### PR DESCRIPTION
**Pull request by einheber**

### CODE REVIEW

File: spec/acceptance/pkg_facet_configure_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+require 'spec_helper_acceptance'
+require 'pry'
+
+describe 'Solaris pkg_facet provider' do
+
+  # a manifest to add a facet
+  pp = <<-EOP
```
Comment (shawnferry) [inline spec/acceptance/pkg_facet_configure_spec.rb]: I would lean towards using more descriptive names that pp and pp2 as below.

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -3,19 +3,19 @@
 
 hosts.each do |host|
   # Install Puppet
-#  on host, "pkg install puppet puppet-19 ruby-19"
+  on host, "pkg install puppet puppet-19 ruby-19"
```
Comment (shawnferry) [inline spec/spec_helper_acceptance.rb]: The cross package dependencies should pull everything in are puppet-19 and ruby-19 actually required to make this work?

Comment (einheber) [inline spec/spec_helper_acceptance.rb]: I'll investigate.

**Comment by shawnferry** [comment]
I have no show stopper issues with these as written. If the explicit -RUBYVER installs can be removed it would be preferred as makes the test more future proof.

**Comment by einheber** [comment]
going to redo this.

**Pull request closed without merging**